torchtune/modules/model_fusion/_fusion.py

```diff
@@ -293,27 +314,46 @@ class DeepFusionModel(nn.Module):
         >>> output = model(tokens, mask, encoder_input, encoder_mask, input_pos)
 
     Args:
-        decoder (TransformerDecoder): decoder module
+        decoder (torchtune.modules.TransformerDecoder): decoder module
         encoder (nn.Module): encoder module
     """
 
     def __init__(
         self,
-        decoder: TransformerDecoder,
+        decoder: torchtune.modules.TransformerDecoder,
```

Comment:
had to do this due to circular imports, open to something more sane looking

Reply:
I'm confused, what's the circular import here? Seems to me like TransformerDecoder should not be depending on fusion, right?